kdc/pkinit.c

```diff
@@ -230,6 +230,7 @@ generate_dh_keyblock(krb5_context context,
 	    size -= dh_gen_keylen;
 	    memmove(dh_gen_key + size, dh_gen_key, dh_gen_keylen);
 	    memset(dh_gen_key, 0, size);
+	    dh_gen_keylen += size;
```

Comment:
Does the buggy client mean we need to try both possibilities now?

Reply:
Yes, if we want to interoperate with buggy Heimdal clients without sporadic failures. For the other way round, interoperating with buggy Heimdal servers, the client probably would need to keep regenerating the key until it got a working one.

Reply:
Back in 2010 there were interop problems with MIT which resulted in the submission of both 7639f83561a5d8967094e7e76caebb6d1890791f and then 6e05462c1e83a7dc02995430dffb4213132fe373.  Does the existing code successfully interoperate with MIT?

Reply:
>  Does the existing code successfully interoperate with MIT?

I don’t know. I haven’t tested that.